document-store/src/main/java/org/hypertrace/core/documentstore/expression/RelationalExpression.java

```diff
@@ -0,0 +1,22 @@
+package org.hypertrace.core.documentstore.expression;
+
+import lombok.Value;
+
+/**
+ * Expression representing a condition for filtering
+ *
+ * <p>Example: <code>
+ *     company IN ('Traceable', 'Harness')
+ *  </code> can be constructed as <code>
+ *     RelationalExpression.of(
+ *         LiteralExpression.of("company"),
+ *         RelationalOperator.IN,
+ *         ConstantExpression.ofStrings("Traceable", "Harness"))));
+ * </code>
+ */
+@Value(staticConstructor = "of")
+public class RelationalExpression implements Filterable {
```

Comment:
would `company EXISTS` be a relational expression? If we don't need it, can omit - but we should ensure that adding it would not change the design.